app/presenters/candidate_interface/application_form_presenter.rb

```diff
@@ -108,6 +108,10 @@ def contact_details_completed?
       @application_form.contact_details_completed
     end
 
+    def contact_details_valid?
+      ContactDetailsForm.build_from_application(@application_form).valid?
```

Comment:
This actually needs to be:

```suggestion
      form = ContactDetailsForm.build_from_application(@application_form)
      form.valid?(:base) && form.valid?(:address) && form.valid?(:address_type)
```